SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/PartitionManager.cpp

```diff
@@ -34,18 +36,52 @@ CommandRequest CreateClaimPartitionRequest(Worker_PartitionId Partition)
 }
 } // namespace
 
-FPartitionManager::Impl::Impl(const FSubView& InServerWorkerView, ViewCoordinator& Coordinator, TUniquePtr<InterestFactory>&& InInterestF)
+PartitionInterestUpdate::PartitionInterestUpdate()
+{
+	const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>();
+	if (Settings->PartitionSystemClass)
+	{
+		USpatialPartitionSystem* PartitionSystem = Cast<USpatialPartitionSystem>(Settings->PartitionSystemClass->GetDefaultObject());
+		TArray<FLBDataStorage*> DataStores = PartitionSystem->GetData();
+		for (auto Store : DataStores)
+		{
+			PartitionMetaData.Append(Store->GetComponentsToWatch().Array());
+		}
+	}
+}
+
+void PartitionInterestUpdate::Clear()
+{
+	NewWorker.Empty();
+	NewServerWorker.Empty();
+	NewPartition.Empty();
+	RemovedWorker.Empty();
+	RemovedServerWorker.Empty();
+	RemovedPartition.Empty();
+}
+
+bool PartitionInterestUpdate::HasChanged() const
+{
+	return NewPartition.Num() > 0 || NewWorker.Num() > 0 || NewServerWorker.Num() > 0 || RemovedPartition.Num() > 0
+		   || RemovedWorker.Num() > 0 || RemovedServerWorker.Num() > 0;
+}
+
+FPartitionManager::Impl::Impl(const FSubView& InServerWorkerView, ViewCoordinator& Coordinator, InterestFactory& InInterestF)
 	: WorkerView(InServerWorkerView)
 	, SystemWorkerView(Coordinator.CreateSubView(SpatialConstants::WORKER_COMPONENT_ID, SpatialGDK::FSubView::NoFilter,
 												 SpatialGDK::FSubView::NoDispatcherCallbacks))
 	, PartitionView(Coordinator.CreateSubView(SpatialConstants::PARTITION_ACK_COMPONENT_ID, SpatialGDK::FSubView::NoFilter,
 											  SpatialGDK::FSubView::NoDispatcherCallbacks))
-	, InterestF(MoveTemp(InInterestF))
+	, InterestF(InInterestF)
```

Comment:
Intentionally no more move?

Reply:
Yep - held by the FSpatialStrategySystem now - which owns the PartitionManager.